openreview/tools.py

```diff
@@ -146,8 +146,14 @@ def get_profiles(client, ids_or_emails, with_publications=False, as_dict=False):
         profiles.extend(batch_profiles)
 
     if as_dict:
-        for i, profile in enumerate(profiles):
-            profiles_as_dict[ids[i]] = profile
+        for profile in profiles:
+            for name in profile.content.get("names", {}):
+                if name.get("username") in ids:
+                    profiles_as_dict[name.get("username")] = profile
+                    continue
```

Comment:
If we pass two different tilde ids that belongs to the same profile we are going to get a profile for one tilde id and None for the other. I think you can reproduce this adding more tests. I believe there are profiles with alternate names. 

You must check that all the `ids` have a profile. I would suggest to do the following:

```
profiles_by_name = {}
for profile in profiles:
      for name in profile.content.get('names', []):
            if name.get('username'):              
                  profiles_by_name[name.get('username')] = profile

for id in ids:
      profiles_as_dict[id] = profile_by_name.get(id)
```

Reply:
Emails will have same issue

Reply:
could you provide a test? that is the best way to verify if it fails.